lib/helper.js

```diff
@@ -71,5 +71,18 @@ module.exports = {
       }
     }
     return null;
+  },
+  /**
+   * Find the original caller that generate the error.
+   * @returns {*}
+   */
+  getServiceNameFromStackTrace: function() {
+    const stack = new Error().stack.toString();
+    const stackArr = stack.split(' at ');
+    const callers = stackArr.filter(trace => {
+      return /node-sdk/.test(trace) && !/node_modules/.test(trace) && !/anonymous/.test(trace) && trace;
+    });
+
+    return callers[callers.length - 1].split(' (')[0];
```

Comment:
@adrien2p Yea, I think you're right. I'll kill this one for now and try to find time to roll out a better version of this + some of the other improvements that are currently sitting on the back burner.